obspy/io/nordic/core.py

```diff
@@ -0,0 +1,1379 @@
+# -*- coding: utf-8 -*-
+"""
+Nordic file format support for ObsPy
+
+:copyright:
+    The ObsPy Development Team (devs@obspy.org)
+:license:
+    GNU Lesser General Public License, Version 3
+    (https://www.gnu.org/copyleft/lesser.html)
+
+.. note::
+
+    We have not implemented any handling of focal mechanism solutions between
+    the two formats.
+
+.. note::
+
+    Pick time-residuals are handled in event.origins[0].arrivals, with
+    the arrival.pick_id linking the arrival (which contain calculated
+    information) with the pick.resource_id (where the pick contains only
+    physical measured information).
+"""
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+from future.builtins import *  # NOQA @UnusedWildImport
+
+import warnings
+
+import numpy as np
+import os
+
+from obspy import UTCDateTime
+from obspy.core.event import Event, Origin, Magnitude, Comment, Catalog
+from obspy.core.event import EventDescription, CreationInfo
+from obspy.core.event import Pick, WaveformStreamID, Arrival, Amplitude
+
+
+def is_sfile(sfile):
+    """
+    Basic test of whether the file is nordic format or not.
+
+    Not exhaustive, but checks some of the basics.
+
+    :type sfile: str
+    :param sfile: Path to sfile
+    :rtype: bool
+    """
+    if not hasattr(sfile, "readline"):
+        try:
+            with open(sfile, 'r') as f:
+                topline = f.readline()
+                if not len(topline.rstrip()) == 80:
+                    return False
+                f.seek(0)
+                for line in f:
+                    if line[79] in [' ', '1']:
+                        head_line = line
+                        break
+        except:
+            return False
+    else:
+        topline = sfile.readline()
+        if not len(topline.rstrip()) == 80:
+            return False
+        sfile.seek(0)
+        for line in sfile:
+            if line[79] in [' ', '1']:
+                head_line = line
+                break
+    if 'head_line' in locals():
+        try:
+            sfile_seconds = int(head_line[16:18])
+        except ValueError:
+            return False
+        if sfile_seconds == 60:
+            sfile_seconds = 0
+            add_seconds = 60
+        else:
+            add_seconds = 0
+        try:
+            UTCDateTime(int(head_line[1:5]), int(head_line[6:8]),
+                        int(head_line[8:10]), int(head_line[11:13]),
+                        int(head_line[13:15]), sfile_seconds,
+                        int(head_line[19:20]) * 100000) + add_seconds
```

Comment:
The logic with the 60 seconds is also needed later on - maybe encapsulate in a function?